### PR TITLE
add favorite place using coredata

### PR DESCRIPTION
**Pull request by tungday**
## Related Tickets
- [#Task 78868: Content](https://edu-redmine.sun-asterisk.vn/issues/78868)
## WHAT
- add and delete favorite place using coredata
## Evidence (Screenshot or Video)
image

image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: iMap/iMap/Controllers/PlaceTableViewController.swift [hunk 1]
```diff
@@ -13,15 +14,61 @@ final class PlacesTableViewController: UITableViewController {
...
+            } else {
+                for object in results {
+                    context.delete(object)
+                }
+                try context.save()
+            }
+        } catch {
+            print("Failed to update favorite status: \(error)")
+        }
+    }
```
Comment (duynv-3452) [inline iMap/iMap/Controllers/PlaceTableViewController.swift:69]: liên quan tới Coredata thì e viết riêng vào 1 file CoreDataService nhé

File: iMap/iMap/Controllers/PlaceTableViewController.swift [hunk 2]
```diff
@@ -6,55 +6,85 @@ final class PlacesTableViewController: UITableViewController {
...
+        
+        cell.contentView.subviews.filter { $0 is UIButton }.forEach { $0.removeFromSuperview() }
+        
+        cell.contentView.addSubview(favoriteButton)
+        NSLayoutConstraint.activate([
+            favoriteButton.trailingAnchor.constraint(equalTo: cell.contentView.trailingAnchor, constant: -16),
+            favoriteButton.centerYAnchor.constraint(equalTo: cell.contentView.centerYAnchor),
+            favoriteButton.widthAnchor.constraint(equalToConstant: 24),
+            favoriteButton.heightAnchor.constraint(equalToConstant: 24)
+        ])
```
Comment (tuta-0902) [inline iMap/iMap/Controllers/PlaceTableViewController.swift:72]: sao chỗ này k dùng storyboard á

File: iMap/iMap/CoreDataService/CoreDataService.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import Foundation
+import CoreData
+import MapKit
+import UIKit
+
+class CoreDataHelper {
```
Comment (tuta-0902) [inline iMap/iMap/CoreDataService/CoreDataService.swift:6]: final

Comment (tuta-0902) [reply iMap/iMap/CoreDataService/CoreDataService.swift:6]: thử research singleton pattern cho case này nha

**Pull request merged**